src/nupic/types/Sdr.hpp

```diff
@@ -463,6 +463,18 @@ class SparseDistributedRepresentation : public Serializable
 
     void addNoise(Real fractionNoise, Random &rng);
 
+    /**
+     * Modify the SDR by setting a fraction of the bits to zero.
+     *
+     * @param fraction The fraction of bits to set to zero.  Must be between 0
+     * and 1 (inclusive).
+     *
+     * @param seed The seed for the random number generator to draw from.  If not
+     * given, this uses the magic seed 0.  Use the same seed to consistently
+     * kill the same cells.
+     */
+    void killCells(Real fraction, UInt seed=0u);
```

Comment:
I'm still thinking how to make this do both ways: 
- "killing" by turning cells to OFF
- "lighting" by turning ON

We have: 
- `randomize` : achieves given sparsity
- `addNoise` : moves ON bits. Suggest renaming to "shuffle"? 
- `killCells` : suggest renaming to `addNoise(fraction, seed=0, mode={ON,OFF,BOTH}) `

what do you think?

Reply:
maybe better: killCells -> applyMask(frac, seed=0, mode=enum{ON, OFF, TOGGLE})

Reply:
I don't want to change the names of these methods.  All three of these methods do subtly different things, and also I like the current names.  

Randomize does not just add 1's, first it zeros the SDR.  AddNoise and killCells don't zero the SDR first.  So randomize doesn't fit in here.